File: 290. Word Pattern/solution.go
import "strings"

func wordPattern(pattern string, str string) bool {
	s := strings.Split(str, " ")
	l := len(pattern)
	if l != len(s) {
		return false
	}
	for i := 0; i < l; i++ {
		for j := i; j < l; j++ {
			if (pattern[i] == pattern[j] && s[i] == s[j]) || (pattern[i] != pattern[j] && s[i] != s[j]) {
				continue
			} else {
				return false
			}
		}
	}
	return true
}

File: 290. Word Pattern/solution.js
/**
 * @param {string} pattern
 * @param {string} str
 * @return {boolean}
 */
var wordPattern = function(pattern, str) {
  const mapPat = {};
  const mapStr = {};
  str = str.split(" ");

  if (str.length === pattern.length) {
    for (let i = 0; i < str.length; i++) {
      if (
        mapPat[pattern[i]] === str[i] ||
        (!mapPat[pattern[i]] && !mapStr[str[i]])
      ) {
        mapPat[pattern[i]] = str[i];
        mapStr[str[i]] = pattern[i];
        continue;
      }
      return false;
    }
    return true;
  }
  return false;
};
